### PR TITLE
fix: incorrect image release tags in build

### PR DESCRIPTION
**Pull request by johnson2427**
### What I did

The tag that is created when merged to `main` is `main`. We want this to be `latest` and then on version release, we want this to be `stable`

fixes: APE-1769

### How I did it

Added the tag to the build yaml

### How to verify it

After merge, check the release

### Checklist

- [x] All changes are completed
~~- [ ] New test cases have been added~~
~~- [ ] Documentation has been updated~~

### CODE REVIEW

**Comment by antazoey** [comment]
I am guessing this is container-related?

**Comment by johnson2427** [comment]
> I am guessing this is container-related?

yes it is, not getting the right tags on merge to main

**Comment by johnson2427** [comment]
These changes **should** do what we're looking for. I'm not familiar with tagging in github, so I'm riding on documentation, but still a little unclear on this

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -34,10 +34,10 @@ jobs:
           tags: |
             type=ref,event=pr
             type=ref,event=branch,branch=main,latest=true
-            type=ref,event=branch,branch=main
+            type=ref,event=branch,branch=main,tag=latest
             type=ref,event=branch,branch!=main
-            type=semver,pattern={{version}},branch=main,latest=false
-            type=ref,event=tag
+            type=semver,pattern={{version}},branch=main,latest=false,tag=stable
+            type=ref,event=tag,tag={{ tag }}
```
Comment (fubuloubu) [inline .github/workflows/build.yaml:40]: Not quite familiar with how this works

**Review by fubuloubu — DISMISSED** [review]

**Comment by fubuloubu** [comment]
Noticed that the untagged versions were getting, did https://github.com/ApeWorX/ape/pull/2228/commits/a597ba9d13b9e946615df747fb5789501ad8254d resolve that?

**Comment by fubuloubu** [comment]
Also can you delete this?
https://github.com/orgs/ApeWorX/packages/container/ape/261405854?tag=main

**Comment by johnson2427** [comment]
> Noticed that the untagged versions were getting pushed from this PR, did [a597ba9](https://github.com/ApeWorX/ape/commit/a597ba9d13b9e946615df747fb5789501ad8254d) resolve that?
> 
> ref: https://github.com/ApeWorX/ape/pkgs/container/ape/261370690

I'm reviewing docs right now, this isn't supposed to push unless it's merged to main

**Comment by fubuloubu** [comment]
Lastly, would love it if the untagged versions didn't get published at all, feel like that bloats things

File: .github/workflows/build.yaml
Comment (fubuloubu) [inline .github/workflows/build.yaml:1]: I think what you could do for releases is run an extra optional step (after both `-slim` and the normal) to make sure whatever image hash that `stable` gets built as is also tagged with the corresponding `v*` tag as well

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -32,12 +32,9 @@ jobs:
         with:
           images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
           tags: |
-            type=ref,event=pr
-            type=ref,event=branch,branch=main,latest=true
-            type=ref,event=branch,branch=main
-            type=ref,event=branch,branch!=main
-            type=semver,pattern={{version}},branch=main,latest=false
-            type=ref,event=tag
+            type=ref,event=pr,pattern=test
```
Comment (fubuloubu) [inline .github/workflows/build.yaml:35]: what does this do? just a random non-committed tag for use inside the workflow?

**Review by fubuloubu — APPROVED** [review]

File: .github/workflows/build.yaml (hunk 2 above)
Comment (johnson2427) [reply .github/workflows/build.yaml]: just trying to see what results I get when I change things so I know what it'll look like when we merge

**Comment by johnson2427** [comment]
> Lastly, would love it if the untagged versions didn't get published at all, feel like that bloats things

you don't want to release unless it's a tagged version? So don't release on merge to main?

**Comment by fubuloubu** [comment]
> > Lastly, would love it if the untagged versions didn't get published at all, feel like that bloats things
> 
> you don't want to release unless it's a tagged version? So don't release on merge to main?

No no, those random untagged releases which are image hashes. Don't know where those come from

**Comment by johnson2427** [comment]
> > > Lastly, would love it if the untagged versions didn't get published at all, feel like that bloats things
> > 
> > 
> > you don't want to release unless it's a tagged version? So don't release on merge to main?
> 
> No no, those random untagged releases which are image hashes. Don't know where those come from

I *think* this is resolved. I don't see it in the package releases. And I'm no longer pushing until we merge to main. We should be okay here

File: .github/workflows/build.yaml
Comment (johnson2427) [reply .github/workflows/build.yaml:1]: To be clear, we'd want a `stable` and `vX.X.X` release on a version tag release

**Comment by fubuloubu** [comment]
> > > > Lastly, would love it if the untagged versions didn't get published at all, feel like that bloats things
> > > 
> > > 
> > > you don't want to release unless it's a tagged version? So don't release on merge to main?
> > 
> > No no, those random untagged releases which are image hashes. Don't know where those come from
> 
> I *think* this is resolved. I don't see it in the package releases. And I'm no longer pushing until we merge to main. We should be okay here

So we can get this merged to see?

**Review by fubuloubu — APPROVED** [review]

File: .github/workflows/build.yaml
Comment (fubuloubu) [reply .github/workflows/build.yaml:1]: Yes that's correct. The stable always points at the same image as the latest tagged release, but if you use a release version then it never changes

Comment (johnson2427) [reply .github/workflows/build.yaml:1]: np, let me add this

**Comment by johnson2427** [comment]
If this does not work, I'm removing the metadata step and forcefully injecting the tags

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**